File: app/src/main/java/com/cellblock70/popularmovies/data/database/MovieDatabase.java
package com.cellblock70.popularmovies.data.database;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.migration.Migration;
import androidx.sqlite.db.SupportSQLiteDatabase;

@Database(entities = {Movie.class, MovieReview.class, MovieTrailer.class},
          version = 2, exportSchema = false)
public abstract class MovieDatabase extends RoomDatabase {

    public abstract MovieDao movieDao();

    static final Migration MIGRATION_2 = new Migration(1, 2) {
        @Override
        public void migrate(@NonNull SupportSQLiteDatabase database) {

        }
    };

    private static volatile MovieDatabase INSTANCE;

    public static MovieDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (MovieDatabase.class) {
                if (INSTANCE == null) {
                    // Create database here
                    INSTANCE = Room.databaseBuilder(context, MovieDatabase.class, "movie.db").build();
                }
            }
        }
        return INSTANCE;
    }

}

File: app/src/main/java/com/cellblock70/popularmovies/data/database/MovieDatabase.kt
package com.cellblock70.popularmovies.data.database

import androidx.room.Database
import androidx.room.RoomDatabase

@Database(
    entities = [MovieTrailer::class, Favorite::class, MovieReview::class, Movie::class],
    version = 2,
    exportSchema = false
)
abstract class MovieDatabase : RoomDatabase() {

    abstract fun movieDao(): MovieDao
}
